runtime/runtime.c

```diff
@@ -253,9 +253,35 @@ void* allocate_n_bytes_ptr(u32 n) {
 WEAK void populate_globals() {}
 
 // Code that actually runs the wasm code
-IMPORT i32 wasmf_main(i32 a, i32 b);
+IMPORT void wasmf__start(void);
+
+int runtime_argc = 0;
+
+char *runtime_argv_buffer = NULL;
+int runtime_argv_buffer_len = 0;
+
+void runtime_cleanup() {
+    // Cancel any pending timeout
+    //    if (wasm_execution_timeout_ms) {
+    //        cancel_timeout();
+    //    }
+
+    free(runtime_argv_buffer);
+    runtime_argv_buffer_len = 0;
+}
 
 int runtime_main(int argc, char** argv) {
+    // Set argc and argv to globals, these are later used by the WASI syscalls 
+    runtime_argc = argc;
+
+    // Copy the null terminated args one after the other into a global with the format WASI calls expect
+    for (int i = 0; i < argc; i++) {
+        size_t arg_len = strlen(argv[i]);
```

Comment:
Is this loop not copying in the `\0`s? If you print out the `runtime_argv_buffer`, do you see all arguments (wrong), or only the first (correct)?

Or are the arguments supposed to be space-separated arguments (in which case we should add a ` ` before the next string).

Reply:
I was originally copying in the null terminators, but I think that's actually wrong. Based on what I see running my wat sample under wasmtime, the backing buffer seems to not be delimited at all, and the vector of base addresses returned by `args_sizes_get` seems to be how it's chopped up. I'll sanity check that with a C example though!